### DISCUSSION TITLE
How to reset a store with one action's method

### DISCUSSION THREAD

**Discussion by Trandx** [vuejs/pinia#1692] [Help and Questions]
Hello guys. 
I want to write one method that can reset my store directly in a auth store file.
`
    import {defineStore} from "pinia"
    export const useAuthStore = defineStore("auth", {
        state: () => ({
            auth:{
                authenticated:false,
                user:{},
                sessionExpireAt: null,
            }
            
        }),
      getters: {
          // check if user is authenticated
          authenticated: (state) => state.auth.authenticated ,
          
          // get user datas
          user(){ return this.auth.user},
  
          // get session expiration date
          sessionExpireAt: (state) => state.sessionExpireAt,
  
          //
          
      },
      actions: {
  
         // add authenticated user
         add (data) {
              this.auth = {...this.auth, ...data}
              return this.auth
         },
  
         // reset store when user logout
         reset() {
           // what can i put there ?
          } 
      }
  })
`

**Comment by Trandx** [comment]
I don't want to write a code like...
`...
reset() {
this.auth = {
            authenticated:false,
            user:{},
            sessionExpireAt: null,
        }
return this.auth
}
`
before our answers, i think that if in next release we have a method like `state.reset()` directly in "defineStore" without the "store.$reset()" it will be very cool !!

**Comment by posva** [comment]
you can just call `this.$reset()` within the `reset()` action
